requirements.txt

```diff
@@ -1,3 +1,4 @@
+functorch
```

Comment:
The main thing that held us back from bumping up the torch dependency was csprng. Sorry, I'm not familiar with the latest take on that, are we okay with opacus sans csprng?

Reply:
> are we okay with opacus sans csprng?

csprng has been an optional dependency since fc23b23ed19a9d0bdc1d1a7a55bcef3a5904b36f (which means you could have functorch+pytorch1.11 without csprng even though it's not ideal). That being said, I think @romovpa fixed the csprng dependency on older version of Pytorch but not 100% sure.

> given that functorch requires torch>=1.11, do we want to update our dependency too?

If we put functorch, wouldn't pip automatically apply this constraint to the torch version?

Reply:
> csprng has been an optional dependency since fc23b23

That's right! IIUC, csprng is optional dependency, but it can still be used if needed. csprng is only supported until torch 1.8, and the current opacus is compatible with torch 1.8.
However, if we change the torch version from `torch>=1.8` to `torch>=1.11` in L3 in this file, opacus will no longer be useable with torch 1.8 and consequently with csprng.
I agree that @romovpa would have the best context here.

Reply:
Just reopened this thread. Did we reach a conclusion on which torch version to support?
Given that functorch is listed as a requirement, what would happen with when someone tries to use functorch on torch 1.8.1?